File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int index = -1;
        int min = Integer.MAX_VALUE;

        for (int i = 0; i < points.length; i++) {
            if( x == points[i][0] || y == points[i][1]){
                int d = Math.abs(x - points[i][0]) + Math.abs(y - points[i][1]);

                if(d<min) {
                    min = d;
                    index = i;
                }
            }
        }

        return index;
    }
}

File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        valid = [p for p in points if p[0] == x or p[1] == y]

        if(len(valid)>0):
            _min = min(valid,key = lambda p: abs(x-p[0])+abs(y-p[1]))
            return points.index(_min)
        else:
            return -1